test/files/run/string-switch.scala

```diff
@@ -0,0 +1,56 @@
+// scalac: -Werror
+import annotation.switch
+import util.Try
+
+object Test extends App {
+
+  def species(name: String) = (name.toLowerCase : @switch) match {
+    case "fido"                     => "dog"
+    case "garfield" | "deuteronomy" => "cat"
+    case "wanda"                    => "fish"
+    case "henry"                    => "horse"
+  }
+  List("fido", "garfield", "wanda", "henry", "felix", "deuteronomy").foreach { n => println(s"$n ${Try(species(n))}") }
+
+  println("=====")
+
+  def collide(in: String) = (in : @switch) match {
+    case "AaAa" => 1
+    case "BBBB" => 2
+    case "cCCc" => 3
+    case x if x == "ddDd" => 4
+  }
+  List("AaAa", "BBBB", "BBAa", "cCCc", "ddDd", "EEee").foreach { s =>
```

Comment:
`assert("AaAa".hashCode == "BBBB".hashCode)` here and near other tests with the same intent.

Reply:
I printed them into the checkfile.